src/topp/OpenSwathWorkflow.cpp

```
@@ -613,6 +621,162 @@ class TOPPOpenSwathWorkflow
+     zeros, negative values, and NaNs/Inf.
+
+    Typical units:
+      - RT window: seconds
+      - m/z window: ppm
+      - IM window: native instrument units
+
+    @param v            The candidate window value.
+    @param min_positive Minimum strictly-positive threshold (default: 1e-9).
+                         Estimates <= this threshold are deemed invalid.
+    @return True if the window is usable; false otherwise.
+  */
+  inline bool is_valid_win(double v, double min_positive = 1e-9) noexcept
+  {
+    return std::isfinite(v) && (v > min_positive);
+  }
+
+  /**
+    @brief Validate, log, and optionally apply an auto-estimated extraction window.
+
+    Behavior:
+      - If @p applicable is false (e.g., no IM data), logs an INFO and leaves @p dst_param unchanged.
+      - If the estimate is invalid, logs a WARN and leaves @p dst_param unchanged.
+      - If the estimate is valid and @p commit is true, logs an INFO and assigns @p dst_param = @p estimate.
+      - If the estimate is valid and @p commit is false, logs an INFO that reports the estimate and that the user value is kept.
+
+    Typical usage:
+      - RT window (seconds)
+      - MS2 m/z window (ppm)
+      - MS1 m/z window (ppm)
+      - IM window (1/k0), only when applicable (e.g., PASEF/IM data)
+
+    @param label       Human-readable label used in logs (e.g., "RT", "MS2 m/z (ppm)", "MS1 ion mobility (1/k0)").
+    @param estimate    Auto-estimated window value to consider.
+    @param dst_param   Destination parameter to update on success (by reference).
+    @param user_value  The current/user-specified value (reported in logs).
+    @param applicable  Whether this window is applicable for the current run/config.
+                       If false, the value is not applied and a note is logged.
+    @param commit      Whether to apply the estimate. If false, only logs the estimate vs. user value.
+                       Default: true (backwards compatible).
+  */
```

Something like this seems like it could be used as a general parameter function in OpenMS as well.

---

The apply_window function

---

mm I agree. Where would it go?

---

maybe have a new function  DefaultParamHandler::validateParams()

Example Usage
```c++
cp.setParameters(new_cp)
new_cp.validateParams() // If parameters are not valid, then set as default
```

Another option is to extend DefaultParamHandler::setParameters() such that if an invalid parameter value is set then it will revert to the default option. You could also add an override to the function if that is more appropriate 

@timosachsenberg what are your thoughts?

---

Hmm not sure if this should be part of the parameter handler. I need to think about this first. Maybe we can discuss at the meeting?

---

In general I think it is much better to fail hard if an invalid parameter is set. But we can look into it… maybe some generic/or consistent way how to tryEstimateOr a parameter …

---

I created a [PR #8205](https://github.com/OpenMS/OpenMS/pull/8205) for updating a value in a Param object for a given key. We can discuss more at the meeting to see if it's an okay approach or if we want to do something else.